### PR TITLE
Speed sections

### PR DESCRIPTION
**Pull request by ManiacMoustache**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Steps to Test
1. Open sim
2. Enable teleop

It didn't crash, I've tested all component methods.

## Changelog

Refer to issue 22

## What?

I've added a Speed Section component, which uses a yml in the configs folder to gather 2D arrays which autonomous motion components use to determine what speed they should use.

## Why?

This allows us to configure speed sections for different robots and keep a central group of speedsections so that we can tune them quickly.

## Automation
Issue: #22

### CODE REVIEW

**Review by hell0w0rld2 — DISMISSED** [review]

File: components/speedSections.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import logging as log
+from utils.configMapper import findConfig
+from utils import yaml
+from robotMap import RobotMap
+import os
+class SpeedSections():
+    configPath = os.path.join(findConfig()[1],"speedSections.yml")
```
Comment (ChrisHirsch) [inline components/speedSections.py:7]: What happens here if findConfig()[1] fails? Can we at least throw a good error about needing the speedSections.yml file?

File: configs/speedSections.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+doof:
```
Comment (ChrisHirsch) [inline configs/speedSections.yml:1]: Should the speedSections be subsection in the bot's config itself? ie the doof speedSections can be in doof.yml and minibot.yml etc? This means less files and more complete config for a specific bot.

**Review by ChrisHirsch — DISMISSED** [review]
Let me know your thoughts on the error handling and moving the bot specific configs into the bot specific yaml instead of a new yaml file

**Review by hell0w0rld2 — APPROVED** [review]
👍

**Review by hell0w0rld2 — APPROVED** [review]
👍

**Review by Leaf-life — APPROVED** [review]
g

**Pull request merged**